usage/usage.go

```diff
@@ -23,20 +24,29 @@ type orgstat struct {
 // tracks for every org
 type Usage struct {
 	sync.Mutex
-	period uint32
-	now    map[int]orgstat
-	prev   map[int]orgstat
-	stop   chan struct{}
+	period    uint32
+	now       map[int]orgstat
+	prev      map[int]orgstat
+	stop      chan struct{}
+	shardID   string
+	partition int32
 }
 
-func New(period uint32, m mdata.Metrics, i idx.MetricIndex, cl clock.Clock) *Usage {
+func New(period uint32, m mdata.Metrics, i idx.MetricIndex, cl clock.Clock, partitions []int32) *Usage {
 	metrics = m
 	metricIndex = i
 	Clock = cl
+	partition := partitions[0]
+	shardID := "shard"
+	for _, p := range partitions {
+		shardID += "-" + strconv.FormatInt(int64(p), 10)
+	}
 	ret := &Usage{
-		period: period,
-		now:    make(map[int]orgstat),
-		stop:   make(chan struct{}),
+		period:    period,
+		now:       make(map[int]orgstat),
+		stop:      make(chan struct{}),
+		shardID:   shardID,
```

Comment:
There is also nothing which sorts the partition IDs after they've been parsed out of the file: https://github.com/raintank/metrictank/blob/master/input/kafkamdm/kafkamdm.go#L147-L164
But considering that the list of partition IDs is part of the shardID i think it might make sense to sort them in order to guarantee that the same combination of partitions results in the same shardID